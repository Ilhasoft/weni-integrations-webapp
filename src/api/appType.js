import qs from 'query-string';
import request from './request';

const resource = '/api/v1/apptypes';
export default {
  getAllAppTypes(filter) {
    const queryString = qs.stringify(filter);
    return request.$http.get(`${resource}/?${queryString}`);
  },
  getAppType(code) {
    return request.$http.get(`${resource}/${code}/`);
  },
  listComments(appCode) {
    return request.$http.get(`${resource}/${appCode}/comments/`);
  },
  createComment(appCode, payload) {
    return request.$http.post(`${resource}/${appCode}/comments/`, payload);
  },
  updateComment(appCode, commentId, data) {
    return request.$http.put(`${resource}/${appCode}/comments/${commentId}/`, data);
  },
  deleteComment(appCode, commentUuid) {
    return request.$http.delete(`${resource}/${appCode}/comments/${commentUuid}/`);
  },
  postRating(appCode, payload) {
    return request.$http.post(`${resource}/${appCode}/ratings/`, payload);
  },
  createApp(appCode, data) {
    return request.$http.post(`${resource}/${appCode}/apps/`, data);
  },
  fetchFeatured() {
    return request.$http.get(`${resource}/featureds/`);
  },
};
