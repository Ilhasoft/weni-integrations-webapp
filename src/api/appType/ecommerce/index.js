import request from '@/api/request.js';

const resource = '/api/v1/apptypes';

export default {
  async getAllEcommerceTypes() {
    return await request.$http.get(`${resource}/?category=ecommerce`).then((r) => r.data);
  },
  connectVtexCatalog(code, appUuid, payload) {
    return request.$http.post(`${resource}/${code}/${appUuid}/catalogs/`, payload);
  },
  async getVtexAppUuid(code) {
    return await request.$http.get(`${resource}/${code}/apps/get-app-uuid/`).then((r) => r.data);
  },
};
