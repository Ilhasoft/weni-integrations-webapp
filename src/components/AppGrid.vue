<template>
  <div>
    <section id="app-grid">
      <p class="app-grid__title">{{ $t(section) }}</p>

      <div class="app-grid__content">
        <unnnic-card
          ref="unnnic-marketplace-card"
          class="app-grid__content__item"
          v-for="app in apps"
          v-bind:key="app.id"
          type="marketplace"
          :title="app.name"
          :description="app.description"
          :icon="app.icon"
          :id="app.id"
          :comments="`${app.comments_count} ${$t('comments')}`"
          :rating="app.rating.average || 0"
          :iconSrc="app.icon"
          :typeAction="type"
          :buttonAction="/* istanbul ignore next */ () => openAddModal(app.id, app.name)"
          clickable
          @openModal="openAppModal(app.code)"
        />
      </div>
    </section>

    <unnnic-modal
      ref="unnnic-add-modal"
      :showModal="showAddModal"
      :text="$t('installed_app_title')"
      scheme="feedback-green"
      modal-icon="check-circle-1-1"
      @close="closeAddModal"
    >
      <span slot="message" v-html="$t('installed_app_description')"></span>
      <unnnic-button
        ref="unnnic-add-modal-close-button"
        slot="options"
        type="terciary"
        @click="closeAddModal"
        >{{ $t('close') }}</unnnic-button
      >
      <unnnic-button
        ref="unnnic-add-modal-navigate-button"
        slot="options"
        type="primary"
        @click="navigateToMyApps()"
      >
        {{ $t('access_my_apps') }}
      </unnnic-button>
    </unnnic-modal>
  </div>
</template>

<script>
  import { mapActions } from 'vuex';
  export default {
    name: 'AppGrid',
    props: {
      section: {
        type: String,
        default: null,
        validator(value) {
          return (
            [
              'communication_channels',
              'attendance_managers',
              'configured_apps',
              'installed_apps',
            ].indexOf(value) !== -1
          );
        },
      },
      type: {
        type: String,
        validator(value) {
          return ['add', 'config', 'edit'].indexOf(value) !== -1;
        },
      },
    },
    data() {
      return {
        showAddModal: false,
        apps: [],
      };
    },
    async mounted() {
      const category = this.getCategory(this.section);
      this.loadApps({ category });
    },
    methods: {
      ...mapActions(['getAllAppTypes']),
      async loadApps(filter) {
        if (this.type == 'add') {
          const { data } = await this.getAllAppTypes(filter);
          this.apps = data;
        } else {
          this.apps = [
            {
              code: 'wwc',
              name: 'Weni Web Chat MOCK',
              description:
                'Ullamco occaecat et id cillum. Amet exercitation nisi amet fugiat mollit minim est. Officia in enim amet ipsum Lorem velit sint pariatur sunt magna cupidatat. Magna non ea qui nisi ut.s',
              summary: 'Ullamco occaecat et id cillum.',
              category: 'channel',
              bg_color: '#00DED333',
              icon: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg',
              assets: [
                {
                  type: 'image_banner',
                  url: 'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/Ilustra%C3%A7%C3%A3o%20banner%201.png',
                  description: '',
                },
              ],
              usersCount: 25,
              rating: {
                average: 4.9,
              },
              commentsCount: 3,
              config: {
                title: 'Chat Title',
                subtitle: 'Chat subtitle',
                placeholder: 'Chat placeholder',
                fullScreenButton: false,
                unreadIndicator: false,
                keepHistory: false,
                avatarImage:
                  'https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/iconfinder_13-whatsapp_4202050+1.svg',
                customCSS: '',
                primaryColor: '#009E96',
                messageDelay: 1,
              },
            },
          ];
        }
      },
      getCategory(section) {
        const categories = {
          communication_channels: 'channel',
          attendance_managers: 'ticket',
          configured_apps: null,
          installed_apps: null,
        };
        return categories[section];
      },
      openAddModal() {
        this.showAddModal = true;
      },
      closeAddModal() {
        this.showAddModal = false;
      },
      navigateToMyApps() {
        this.$router.push('apps');
      },
      openAppModal(code) {
        this.$router.push(`/apps/${code}/details`);
      },
    },
  };
</script>

<style lang="scss" scoped>
  .app-grid {
    &__title {
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-title-sm;
    }

    &__content {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
      grid-gap: $unnnic-spacing-stack-sm;
      align-items: flex-start;
    }
  }
</style>
