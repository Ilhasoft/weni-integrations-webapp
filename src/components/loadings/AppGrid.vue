<template>
  <div>
    <div class="title">
      <unnnic-skeleton-loading tag="div" width="40px" height="40px" />
      <unnnic-skeleton-loading tag="div" width="208px" height="40px" />
    </div>
    <div class="grid">
      <unnnic-skeleton-loading
        v-for="index in 4"
        :key="index"
        class="grid__item"
        tag="div"
        width="256px"
        height="200px"
      />
    </div>
  </div>
</template>

<script>
  // import SkeletonLoading from '../Skeleton/SkeletonLoading.vue';
  export default {};
</script>

<style lang="scss" scoped>
  .title {
    display: flex;
    gap: $unnnic-spacing-inline-xs;
    margin: $unnnic-spacing-stack-md 0;
  }
  .grid {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(256px, 256px));
    grid-gap: $unnnic-spacing-stack-sm;
    align-items: flex-start;
  }
</style>
