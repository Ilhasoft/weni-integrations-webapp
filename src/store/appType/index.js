import state from './state';
import actions from './actions';
import mutations from './mutations';

import channels from './channels';
import comments from './comments';
import externals from './externals';

export default {
  modules: {
    channels,
    comments,
    externals,
  },
  actions,
  mutations,
  state,
};
