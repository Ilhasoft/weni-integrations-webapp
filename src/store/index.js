import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
  },
});

store.dispatch('retriveAuthToken');
store.dispatch('retriveSelectedOrg');
store.dispatch('retriveSelectedProject');

export default store;
