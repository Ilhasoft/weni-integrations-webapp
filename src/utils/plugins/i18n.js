import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
