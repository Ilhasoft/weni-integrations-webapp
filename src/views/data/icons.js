const avatarIcons = {
  channel: {
    icon: 'sms',
    scheme: 'aux-purple-500',
    filled: true,
    opacity: false,
  },
  ticket: {
    icon: 'messaging-we-chat-3',
    scheme: 'aux-blue',
    filled: true,
    opacity: false,
  },
  'bi-tools': {
    icon: 'bar_chart_4_bars',
    scheme: 'aux-red-500',
    filled: true,
    opacity: false,
  },
  external: {
    icon: 'power',
    scheme: 'aux-orange-500',
    filled: true,
    opacity: false,
  },
  configured: {
    icon: 'cog-1',
    scheme: 'aux-purple',
    filled: true,
    opacity: false,
  },
  installed: {
    icon: 'check-circle-1-1',
    scheme: 'aux-blue',
    filled: true,
    opacity: false,
  },
  recommended: {
    icon: 'rating-star-1-1',
    scheme: 'aux-orange',
    filled: true,
    opacity: false,
  },
};

const actionIcons = {
  add: 'add-1',
  config: 'cog-1',
  edit: 'pencil-write-1',
};

const cardIcons = {
  add: 'add-1',
  config: 'navigation-menu-vertical-1',
  edit: 'navigation-menu-vertical-1',
};

export { avatarIcons, actionIcons, cardIcons };
