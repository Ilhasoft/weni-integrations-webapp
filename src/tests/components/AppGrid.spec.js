import { mount } from '@vue/test-utils';
import AppGrid from '../../components/AppGrid/index.vue';
import { describe, expect, it, beforeEach, vi } from 'vitest';
import i18n from '@/utils/plugins/i18n';
import { createTestingPinia } from '@pinia/testing';
import { app_type } from '@/stores/modules/appType/appType.store';
import { avatarIcons } from '@/views/data/icons';
describe('AppGrid', () => {
  let wrapper;
  const pushMock = vi.fn();
  const pinia = createTestingPinia({
    stubActions: false,
    createSpy: vi.fn,
  });
  beforeEach(async () => {
    wrapper = mount(AppGrid, {
      global: {
        plugins: [pinia, i18n],
        mocks: {
          $router: {
            push: pushMock,
          },
          props: {
            apps: [
              {
                code: 'vtex',
                name: 'Vtex',
                description: 'vtex.data.description',
                summary: 'vtex.data.summary',
                category: 'ecommerce',
                icon: null,
                bg_color: null,
                config_design: 'pre-popup',
                rating: { average: null, mine: null },
                comments_count: 0,
                integrations_count: 3,
                metrics: 58602143,
                can_add: false,
                assets: [],
              },
            ],
            avatar: avatarIcons.channel,
          },
        },
      },
    });
    await wrapper.vm.$nextTick();
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('mounted', () => {
    it('should call updateGridSize', async () => {
      const spy = vi.spyOn(AppGrid.methods, 'updateGridSize');
      await wrapper.vm.$nextTick();
      expect(spy).toBeDefined;
    });
  });

  describe('methods', () => {
    it('should reset currentRemoval and showRemoval', async () => {
      const showRemoveModal = wrapper.vm.showRemoveModal;
      await wrapper.vm.toggleRemoveModal();
      expect(wrapper.vm.currentRemoval).toBe(null);
      expect(wrapper.vm.showRemoveModal).toBe(!showRemoveModal);
    });

    it('calls removeApp and toggles remove modal', async () => {
      const store = app_type();
      store.errorDeleteApp = false;

      const spyToggleRemoveModal = vi.spyOn(wrapper.vm, 'toggleRemoveModal');
      const spyDeleteApp = vi.spyOn(store, 'deleteApp').mockResolvedValue();

      await wrapper.vm.removeApp('1234', '5678');

      expect(spyDeleteApp).toHaveBeenCalledWith({ code: '1234', appUuid: '5678' });
      expect(spyToggleRemoveModal).toHaveBeenCalledTimes(1);
    });

    it('should call openAppDetails', async () => {
      await wrapper.vm.openAppDetails('1234');
      expect(pushMock).toHaveBeenCalledWith('/1234/details');
    });
  });
});
