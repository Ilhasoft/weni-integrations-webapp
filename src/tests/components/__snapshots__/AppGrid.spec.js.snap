// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppGrid > matches snapshot 1`] = `
"<div data-v-deb56cf9="">
  <section data-v-deb56cf9="" id="app-grid" maxlength="255" disabled="false">
    <!--v-if-->
    <div data-v-deb56cf9="" class="app-grid__content"></div>
    <!--v-if-->
  </section>
  <unnnic-modal data-v-deb56cf9="" showmodal="false" text="Remove Application" scheme="feedback-red" modal-icon="alert-circle-1"></unnnic-modal>
  <div data-v-7d3d5e15="" data-v-deb56cf9="">
    <transition-stub data-v-7d3d5e15="" name="fade" appear="false" persisted="false" css="true">
      <!--v-if-->
    </transition-stub>
    <unnnic-modal data-v-7d3d5e15="" showmodal="false" text="Cancel modifications" scheme="feedback-yellow" modal-icon="alert-circle-1"></unnnic-modal>
  </div>
</div>"
`;
